src/main/scala/com/dwolla/sqlserver/init/SqlServerDatabaseInitHandlerImpl.scala

```diff
@@ -0,0 +1,199 @@
+package com.dwolla.sqlserver.init
+
+import cats._
+import cats.data._
+import cats.effect.std.Dispatcher
+import cats.effect.{Trace => _, _}
+import cats.syntax.all._
+import com.dwolla.sqlserver.init.InputValidationError.userList
+import com.dwolla.sqlserver.init.SqlServerDatabaseInitHandlerImpl._
+import com.dwolla.sqlserver.init.aws.{ResourceNotFoundException, SecretsManagerAlg}
+import com.dwolla.sqlserver.init.repositories._
+import doobie._
+import doobie.implicits._
+import eu.timepit.refined.{refineMV}
+import feral.lambda.INothing
+import feral.lambda.cloudformation._
+import org.typelevel.log4cats.Logger
+import retry._
+import retry.RetryDetails._
+import retry.syntax.all._
+
+import scala.concurrent.duration.DurationInt
+import scala.util.control.NoStackTrace
+
+class SqlServerDatabaseInitHandlerImpl[F[_] : Temporal : Logger : TransactorFactory](secretsManagerAlg: SecretsManagerAlg[F],
+                                                                                 databaseRepository: DatabaseRepository[ConnectionIO],
+                                                                                 roleRepository: RoleRepository[ConnectionIO],
+                                                                                 userRepository: UserRepository[ConnectionIO],
+                                                                                 loginRepository: LoginRepository[ConnectionIO],
+                                                                                ) extends CloudFormationCustomResource[F, DatabaseMetadata, INothing] {
+
+  val db_reader: RoleName = RoleName(refineMV[SqlServerUserPredicate]("db_datareader"))
```

Comment:
does this smell funny? 😬

Reply:
I think you can use auto-refinement to avoid the `refineMV[SqlServerUserPredicate]` stuff, if that's what you mean